lib/modules/validation.rb

```diff
@@ -0,0 +1,25 @@
+module Validation
+  def validate_name(name)
+    while name.length < 3
+      p 'name is too short,try again'
+      name = gets.chomp
+    end
+    name
+  end
+
+  def validate_move(num, choosed_nums)
+    while num.to_i < 1 || num.to_i > 9 || choosed_nums.include?(num)
+
+      if choosed_nums.include?(num)
+        p "#{num} is already taken, please choose another number"
+      else
+        p 'not a valid number, please choose a number between 1 to 9'
+      end
+
+      num = gets.chomp.strip
+    end
```

Comment:
Move that method into the `Game` class and define it as private
Why? Because, the `choosed_nums` parameter is an element of the Game class.

Instead of 
```rb
num = current_player.make_move(@choosed_nums)
```
AT line 65 of `game.rb`
You will have
```rb
num = this.validation_move(input)
```
`input` is equivalent to **gets.chomp.strip**. Because of the rules `no puts/gets into lib`, this parameter must come from `bin/main.rb`

Into `validation_move`, just use `this.choosed_nums`. The `while` part must also be move into `bin/main.rb`. At the end you will have

```rb
def validate_move(num, choosed_nums)
    if choosed_nums.include?(num)

      if choosed_nums.include?(num)
        return -1
      else
        return -2
      end
    else
      choosed_nums << num
      return num
    end
end
```

Don't forget please, the **while loop** is implemented into `main.rb`
- while loop
- read the input
- call `game.validate_move`. If it returns a positive value, continue, else print an error message and read the move again.